core/string/string_name.cpp

```diff
@@ -148,19 +148,27 @@ void StringName::unref() {
 }
 
 bool StringName::operator==(const String &p_name) const {
-	if (!_data) {
-		return (p_name.length() == 0);
+	if (_data) {
```

Comment:
Could the comparison logic here (and in the other `operator==`) be done in new functions in `_Data`?  This would allow replacing the `get_name()` calls in the `StringName` constructors too.

Reply:
it shouldn't really matter for performance (bc how often are there gonna be hash collisions where it has to check if theyre really equal)
but done anyway